pkg/compact/downsample/downsample.go

```diff
@@ -117,8 +117,8 @@ func Downsample(
 			return id, errors.Wrapf(err, "get series %d", postings.At())
 		}
 
-		for i, c := range chks[1:] {
-			if chks[i].MaxTime >= c.MinTime {
+		for _, c := range chks[1:] {
+			if c.MaxTime >= c.MinTime {
```

Comment:
I think this shouldn't be changed. The logic is different

Reply:
Can you explain it in detail?

Reply:
``` go
		for i, c := range chks[1:] {
			if chks[i].MaxTime >= c.MinTime {
```

Here it is comparing the chks[i].MaxTime and chks[i + 1].MinTime. Not the same chunk.

Reply:
Got it